src/Template/Deserialization.php

```diff
@@ -190,14 +208,27 @@ public function renderAssignDateTimeToField(bool $immutable, string $modelPath,
         ]);
     }
 
-    public function renderAssignDateTimeFromFormat(bool $immutable, string $modelPath, string $jsonPath, string $format): string
+    /**
+     * @param list<string>|string $formats
+     */
+    public function renderAssignDateTimeFromFormat(bool $immutable, string $modelPath, string $jsonPath, array|string $formats, string $timezone = null): string
     {
         $template = $immutable ? self::TMPL_ASSIGN_DATETIME_IMMUTABLE_FROM_FORMAT : self::TMPL_ASSIGN_DATETIME_FROM_FORMAT;
+        $formatVariable = preg_replace_callback(
+            '/([^a-zA-Z]+|\d+)([a-zA-Z])/',
+            static fn ($match) => (ctype_digit($match[1]) ? $match[1] : null).mb_strtoupper($match[2]),
+            $modelPath
+        );
 
         return $this->render($template, [
             'modelPath' => $modelPath,
             'jsonPath' => $jsonPath,
-            'format' => $format,
+            'formats' => array_map(
+                static fn (string $f) => var_export($f, true),
+                \is_string($formats) ? [$formats] : $formats
```

Comment:
i would prefer to do that at the top of this method, immediately sanitize the `$formats`. additionally, we could trigger a deprecation warning when a string is passed instead of an array.

Reply:
i would do it the same as symfony does: https://github.com/symfony/symfony/blob/6c9107334da9e443a4f64f782a93182ac453bf78/src/Symfony/Component/ErrorHandler/DebugClassLoader.php#L337C18-L337C18

`@trigger_error('...', \E_USER_DEPRECATED);`